plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/GoogleApiFactory.java

```diff
@@ -22,31 +22,85 @@
 import com.google.api.client.http.javanet.NetHttpTransport;
 import com.google.api.client.json.JsonFactory;
 import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
 import com.google.api.services.appengine.v1.Appengine;
 import com.google.api.services.appengine.v1.Appengine.Apps;
 import com.google.api.services.cloudresourcemanager.CloudResourceManager;
 import com.google.api.services.cloudresourcemanager.CloudResourceManager.Projects;
 import com.google.cloud.tools.eclipse.util.CloudToolsInfo;
+import com.google.common.annotations.VisibleForTesting;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.Proxy.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.core.net.proxy.IProxyChangeEvent;
+import org.eclipse.core.net.proxy.IProxyChangeListener;
+import org.eclipse.core.net.proxy.IProxyData;
+import org.eclipse.core.net.proxy.IProxyService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 /**
  * Class to obtain various Google cloud Platform related APIs.
- *
- * TODO move this class into a separate API bundle
- * https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1438
  */
-public class GoogleApiFactory {
+@Component
+public class GoogleApiFactory implements IGoogleApiFactory {
 
+  private static final String GOOGLEAPIS_URL = "https://appengine.googleapis.com";
   private static final int DEFAULT_TIMEOUT_MS = 1000;
-  private static final JsonFactory jsonFactory = new JacksonFactory();
-  private static final ConnectionFactory connectionFactory =
-      new TimeoutAwareConnectionFactory(DEFAULT_TIMEOUT_MS, DEFAULT_TIMEOUT_MS);
-  private static final HttpTransport transport =
-      new NetHttpTransport.Builder().setConnectionFactory(connectionFactory).build();
+
+  private static final Logger logger = Logger.getLogger(GoogleApiFactory.class.getName());
+
+  private JsonFactory jsonFactory;
+  private HttpTransport transport;
+
+  private IProxyService proxyService;
+  private String googleApiUrl;
+
+  public GoogleApiFactory() {
+    this(GOOGLEAPIS_URL);
+  }
+
+  @VisibleForTesting
+  GoogleApiFactory(String googleApiUrl) {
+    this.googleApiUrl = googleApiUrl;
+  }
+
+  @Activate
+  public void init() {
+    Preconditions.checkNotNull(proxyService, "proxyService is null");
+    jsonFactory = new JacksonFactory();
+    buildTransport();
+    proxyService.addProxyChangeListener(new IProxyChangeListener() {
```

Comment:
I'm concerned about thread-safety here. It's not a UI widget, so I think this can be executed in any thread.